packages/chaindata-provider/src/ChaindataProvider.ts

```diff
@@ -1,37 +1,700 @@
+  get chains() {
+    return util.wrapObservableWithGetter("Failed to get chains", this.chainsObservable)
+  }
+
+  evmNetworksObservable = new ReplaySubject<(EvmNetwork | CustomEvmNetwork)[]>(1)
+  get evmNetworks() {
+    return util.wrapObservableWithGetter("Failed to get evmNetworks", this.evmNetworksObservable)
+  }
+
+  tokensObservable = new ReplaySubject<Token[]>(1)
+  get tokens() {
+    return util.wrapObservableWithGetter("Failed to get tokens", this.tokensObservable)
+  }
+
+  //
+  // custom item observables
+  //
+
+  get customChainsObservable() {
+    return this.chainsObservable.pipe(map(util.customChainsFilter))
+  }
+  get customChains() {
+    return util.wrapObservableWithGetter("Failed to get custom chains", this.customChainsObservable)
+  }
+
+  get customEvmNetworksObservable() {
+    return this.evmNetworksObservable.pipe(map(util.customEvmNetworksFilter))
+  }
+  get customEvmNetworks() {
+    return util.wrapObservableWithGetter(
+      "Failed to get custom evmNetworks",
+      this.customEvmNetworksObservable
+    )
+  }
+
+  get customTokensObservable() {
+    return this.tokensObservable.pipe(map(util.customTokensFilter))
+  }
+  get customTokens() {
+    return util.wrapObservableWithGetter("Failed to get custom tokens", this.customTokensObservable)
+  }
+
+  //
+  // item ids
+  //
+
+  get chainIdsObservable() {
+    return this.chainsObservable.pipe(map(util.itemsToIds))
+  }
+  get chainIds() {
+    return util.wrapObservableWithGetter("Failed to get chainIds", this.chainIdsObservable)
+  }
+
+  get evmNetworkIdsObservable() {
+    return this.evmNetworksObservable.pipe(map(util.itemsToIds))
+  }
+  get evmNetworkIds() {
+    return util.wrapObservableWithGetter(
+      "Failed to get evmNetworkIds",
+      this.evmNetworkIdsObservable
+    )
+  }
+
+  get tokenIdsObservable() {
+    return this.tokensObservable.pipe(map(util.itemsToIds))
+  }
+  get tokenIds() {
+    return util.wrapObservableWithGetter("Failed to get tokenIds", this.tokenIdsObservable)
+  }
+
+  //
+  // items by id
+  //
+
+  get chainsByIdObservable() {
+    return this.chainsObservable.pipe(map(util.itemsToMapById))
+  }
+  get chainsById() {
+    return util.wrapObservableWithGetter("Failed to get chains by id", this.chainsByIdObservable)
+  }
+
+  get evmNetworksByIdObservable() {
+    return this.evmNetworksObservable.pipe(map(util.itemsToMapById))
+  }
+  get evmNetworksById() {
+    return util.wrapObservableWithGetter(
+      "Failed to get evmNetworks by id",
+      this.evmNetworksByIdObservable
+    )
+  }
+
+  get tokensByIdObservable() {
+    return this.tokensObservable.pipe(map(util.itemsToMapById))
+  }
+  get tokensById() {
+    return util.wrapObservableWithGetter("Failed to get tokens by id", this.tokensByIdObservable)
+  }
+
+  //
+  // items by genesisHash
+  //
+
+  get chainsByGenesisHashObservable() {
+    return this.chainsObservable.pipe(map(util.itemsToMapByGenesisHash))
+  }
+  get chainsByGenesisHash() {
```

Comment:
not super comfortable with async properties, callers may think they can call it multiple times without impact. 
would prefer a getChainsByGenesisHash() method.
wdyt ?

Reply:
I think that's a good idea, I'll change them all to be getXXX methods

Reply:
Done